nir/src/main/scala/scala/scalanative/nir/Vals.scala

```diff
@@ -183,18 +262,37 @@ object Val {
       case _ => false
     }
   }
+
+  /** A heterogeneous collection of data members. */
   final case class StructValue(values: Seq[Val]) extends Val
+
+  /** A homogeneous collection of data members. */
   final case class ArrayValue(elemty: nir.Type, values: Seq[Val]) extends Val
+
+  /** A collection of bytes. */
   final case class ByteString(bytes: Array[scala.Byte]) extends Val {
     def byteCount: scala.Int = bytes.length + 1
   }
+
+  /** A local SSA variable. */
   final case class Local(id: nir.Local, valty: nir.Type) extends Val
+
+  /** A symbol. */
   final case class Global(name: nir.Global, valty: nir.Type) extends Val
 
-  // high-level
+  /** The unit value. */
   case object Unit extends Val
+
+  /** A constant. */
   final case class Const(value: Val) extends Val
+
+  /** A character string. */
   final case class String(value: java.lang.String) extends Val
+
+  /** A virtual value. */
   final case class Virtual(key: scala.Long) extends Val
+
+  /** A class symbol. */
```

Comment:
A reference to `java.lang.Class[_]` of given symbol (`name`). These are emitted as global variables in codegen. `ClassOf` instances are used to deduplicate `Class` instances - there should be only 1 instance per type.

Side note: Even though they're emiited as global variables now, these could be constants. However, when adding multithreading and object monitors we needed capability of editing one of it's fields (lockWord) containing ObjectMonitor object or a bit set of lock word.